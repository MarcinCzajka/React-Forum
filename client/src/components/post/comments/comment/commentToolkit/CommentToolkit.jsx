import React from 'react';
import PropTypes from 'prop-types';
import { Comment, Button } from "semantic-ui-react";
import NewComment from '../../newComment/NewComment';

class CommentToolkit extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            replyContent: '',
            displayTextArea: false
        }
    }

    handleReplyChange = (value) => {
        this.setState({replyContent: value});
    }

    handleReply = () => {
        const { roomId, postId, handleReply } = this.props;

        handleReply(roomId, this.state.replyContent, postId)
            .then(res => {
                this.setState({
                    replyContent: '',
                    displayTextArea: false
                })
            })
    }

    render() {
        const { authorId, userId, postId, removeComment } = this.props;

        return (
            <>
                {this.props.userId ? (
                    <Comment.Actions>
                        <Button size='mini' onClick={() => {this.setState({displayTextArea: true})}}>Reply</Button>
                        {userId === authorId ? <Button size='mini' onClick={() => {removeComment(postId)}}>Delete</Button> : ''}
                    </Comment.Actions>
                ) : ''}
            
                {this.state.displayTextArea ? (
                    <NewComment
                        replyContent={this.state.replyContent}
                        userAvatar={this.context.userAvatar}
                        handleReplyChange={this.handleReplyChange}
                        handleReplyToPost={this.handleReply}
                    />
                ) : ''}
            </>
        )
    }
}

CommentToolkit.propTypes = {
    roomId: PropTypes.string.isRequired,
    postId: PropTypes.string.isRequired,
    userAvatar: PropTypes.string,
    userId: PropTypes.string,
    authorId: PropTypes.string.isRequired,
    handleReply: PropTypes.func.isRequired,
    removeComment: PropTypes.func.isRequired
}

export default CommentToolkit