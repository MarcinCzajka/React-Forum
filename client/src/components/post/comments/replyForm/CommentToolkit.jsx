import React from 'react';
import PropTypes from 'prop-types';
import { Comment, Form, Button } from "semantic-ui-react";

class CommentToolkit extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            textAreaContent: '',
            displayTextArea: false
        }
    }

    resetState = () => {
        this.setState({
            textAreaContent: '',
            displayTextArea: false
        })
    }

    handleReply = () => {
        const { roomId, postId, handleReply } = this.props;

        handleReply(roomId, this.state.textAreaContent, postId)
            .then(this.resetState)
    }

    render() {
        const { authorId, userId, postId, removeComment } = this.props;

        return (
            <>
                {this.props.userId ? (
                    <Comment.Actions>
                        <Button size='mini' onClick={() => {this.setState({displayTextArea: true})}}>Reply</Button>
                        {userId === authorId ? <Button size='mini' onClick={() => {removeComment(postId)}}>Delete</Button> : ''}
                    </Comment.Actions>
                ) : ''}
            
                {this.state.displayTextArea ? (
                    <Form reply>
                        <Form.TextArea 
                            value={this.state.textAreaContent} 
                            onChange={e => this.setState({textAreaContent: e.target.value})} 
                        />
                        <Button
                            onClick={this.handleReply}
                            content='Add Reply'
                            labelPosition='left'
                            icon='edit'
                            primary
                        />
                    </Form>
                ) : ''}
            </>
        )
    }
}

CommentToolkit.propTypes = {
    roomId: PropTypes.string.isRequired,
    postId: PropTypes.string.isRequired,
    userId: PropTypes.string,
    authorId: PropTypes.string.isRequired,
    handleReply: PropTypes.func.isRequired,
    removeComment: PropTypes.func.isRequired
}

export default CommentToolkit