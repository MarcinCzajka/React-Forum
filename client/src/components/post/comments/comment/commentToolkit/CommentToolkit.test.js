import React from 'react';
import { shallow } from 'enzyme';
import { create } from "react-test-renderer";
import CommentToolkit from './CommentToolkit';

const requiredProps = {
    roomId: 'abcde12345',
    postId: 'bcdef23456',
    authorId: 'cdefg34567',
    handleReply: () => {},
    removeComment: () => {}
}

describe('PostListPagination component', () => {
    test('renders without crashing', () => {
        shallow(<CommentToolkit {...requiredProps} />);
    });

    test('match snapshot 1/1', () => {
        const component = create(<CommentToolkit {...requiredProps} />);
        expect(component).toMatchSnapshot();
    });

})