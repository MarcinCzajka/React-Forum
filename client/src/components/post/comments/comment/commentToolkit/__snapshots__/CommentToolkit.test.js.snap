// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListPagination component match snapshot 1/1 1`] = `
Array [
  "",
  "",
]
`;
